databricks/koalas/indexes.py

```diff
@@ -2681,15 +2668,19 @@ def _is_monotonic_increasing(self):
         prev = F.lag(scol, 1).over(window)
 
         cond = F.lit(True)
+        has_not_null = F.lit(True)
         for field in self.spark.data_type[::-1]:
             left = scol.getField(field.name)
             right = prev.getField(field.name)
             compare = MultiIndex._comparator_for_monotonic_increasing(field.dataType)
+            # Since pandas 1.1.4, null value is not allowed at any levels of MultiIndex.
+            # Therefore, we should check `has_not_null` over the all levels.
+            has_not_null = has_not_null & left.isNotNull() & right.isNotNull()
             cond = F.when(left.eqNullSafe(right), cond).otherwise(
                 compare(left, right, spark.Column.__gt__)
             )
 
-        cond = prev.isNull() | cond
+        cond = prev.isNull() | (has_not_null & cond)
```

Comment:
On second thought, it would be:

```diff
diff --git a/databricks/koalas/indexes.py b/databricks/koalas/indexes.py
index 6c09932..37c0945 100644
--- a/databricks/koalas/indexes.py
+++ b/databricks/koalas/indexes.py
@@ -2675,12 +2675,12 @@ class MultiIndex(Index):
             compare = MultiIndex._comparator_for_monotonic_increasing(field.dataType)
             # Since pandas 1.1.4, null value is not allowed at any levels of MultiIndex.
             # Therefore, we should check `has_not_null` over the all levels.
-            has_not_null = has_not_null & left.isNotNull() & right.isNotNull()
+            has_not_null = has_not_null & left.isNotNull()
             cond = F.when(left.eqNullSafe(right), cond).otherwise(
                 compare(left, right, spark.Column.__gt__)
             )

-        cond = prev.isNull() | (has_not_null & cond)
+        cond = has_not_null & (prev.isNull() | cond)

         internal = InternalFrame(
             spark_frame=self._internal.spark_frame.select(
```

Otherwise,

```py
>>> kdf = ks.DataFrame({'a': list('a'), 'b': [1]})
>>> kdf['a'] = None
>>> kdf['b'] = None
>>> kdf.set_index(['a', 'b']).index.is_monotonic
True
>>> kdf.set_index(['a', 'b']).index.to_pandas().is_monotonic
False
```

Reply:
Thanks! addressed the comment and added the related test. :)